src/d3d11/d3d11_shader.h

```diff
@@ -98,7 +98,7 @@ namespace dxvk {
     }
     
     void STDMETHODCALLTYPE GetDevice(ID3D11Device **ppDevice) final {
-      *ppDevice = m_device.ref();
+      *ppDevice = (ID3D11Device*)m_device.ref();
```

Comment:
Why is this necessary? D3D11Device inherits from ID3D11Device, so the pointer conversion should be implicit.

Also, formatting - we don't use C-Style casts.

Reply:
clang 10 shows this error: assigning to 'ID3D11Device *' from incompatible type 'dxvk::D3D11Device *'. So, `static_cast`?

Reply:
Clang shows this error: assigning to `ID3D11Device *` from incompatible type `dxvk::D3D11Device *`. Edited to use `reinterpret_cast`. Tested with latest llvm-mingw toolchain as of this writing.

Reply:
Should be ``static_cast``-able.

Reply:
By that, I mean if it isn't your compiler is broken :p

Reply:
With `static_cast`, even mingw-w64-gcc 10.1.0 shows this error: invalid `static_cast` from type `dxvk::D3D11Device*` to type `ID3D11Device*`.

Reply:
Having to use `reinterpret_cast` implies that the cast is unsafe, and this isn't really acceptable. We should look into a proper fix.

Reply:
Actually, yeah - this happens because the definition of `D3D11Device` is not known at the time. Not sure if compiler bug or our bug, but there doesn't seem to be a clean way to resolve this since `D3D11Device` also needs the shader include.

Reply:
https://github.com/doitsujin/dxvk/commit/56fe52ca3cff3e1cecf8aeef012929abe44bd432 should resolve this without any funky casts.